src/app/services/server-authentication.service.ts

```diff
@@ -0,0 +1,33 @@
+import { Injectable } from '@angular/core';
+import { ServerApiService } from './server-api.service';
+import { Observable } from 'rxjs';
+import { Response } from '../../../api-objects/GenericResponse'
+import { LoginSession } from 'api-objects/LoginSession';
+import { tap } from 'rxjs/operators';
+import { Utilities } from '../common/utlilities';
+import { map } from 'rxjs/operators';
+import { authorityTypes } from '../../../api-objects/UsersObject';
+
+@Injectable()
+export class ServerAuthenticationService {
+    //constructor(private serverApiService: ServerApiService) { }
+    constructor(private serverApiService: ServerApiService) { }
+
+    public login(username: string, password: string): Observable<null> {
+        return this.serverApiService.post<LoginSession>('login', { 'username': username, 'password': password })
+            .pipe(
+                tap((response) => {
```

Comment:
Not an issue, but does this have to be in a tap? Aren't taps meant to be something you do on the side (like logging output for example) while you go through your main process?

Reply:
This is an appropriate use of tap. Tap is used to observe the data of an object without actually changing and values, which is what I'm doing in this context. However, I could technically add it to the map call without any loss of functionality.